src/main/resources/static/common/js/category.js

```diff
@@ -142,18 +142,30 @@ app.controller('categoryController', function($rootScope, $http, $location, $fil
             checkInventory(cartItem.realInStock,itemId,cartItem.quantity);
             $interval(function(){category.loadingAnimation = false;category.showQuantity = true;},500,1);
         }).catch(function(result) {
```

Comment:
@rwang-parasoft I think maybe we can improve this logic here LATER.
Maybe share the shopping cart items info in rootScope instead of making call to get information.

Reply:
> Maybe share the shopping cart items info in rootScope instead of making call to get information.
I think it's not good, since user may open mutiple tabs to add item into cart, so the cart items info in rootScope will be stale.

> At least it's weird to see that my call to get cart items failed because the item is not found, but I still make another call to get detail for the item. Maybe switch the order to get item details first make more sense.
Yes , it maks mor sense. Our design for `/proxy/v1/cartItems/{itemId}` response is not good, we can have a discussion first, then if it's necessary we can refactor it and the order of api call in a new task.